pages/en/lb4/Creating-components.md

```diff
@@ -307,3 +307,297 @@ class EmailComponent {
+      } catch (err) {
+        // You'll probably want more robust error handling in a real server!
+        console.error(err);
+        response.statusCode = 500;
+        response.send(err);
+      }
+    });
+    // Use our router!
+    this.server.expressServer.use(this.routing);
+  }
+}
+```
+
+In addition to the server, the example will include a small type definition for
+convenience.
+
+{% include code-caption.html content="/src/server/rpc-server.ts" %}
+```ts
+import {inject, Context} from '@loopback/context';
+import {Server, Application, CoreBindings} from '@loopback/core';
+import {RPCRouter} from './rpc-router';
+import * as express from 'express';
+import * as http from 'http';
+
+export class RPCServer extends Context implements Server {
+  _server: http.Server;
+  expressServer: express.Application;
+  router: RPCRouter;
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) public app?: Application,
+    @inject('rpcServer.config') public config?: RPCServerConfig
+  ) {
+    super(app);
+    this.config = config || {};
+  }
+
+  async start(): Promise<void> {
+    this.expressServer = express();
+    this.router = new RPCRouter(this);
+    // Currently, there is a limitation with core node libraries
+    // that prevents the easy promisification of some functions, like
+    // http.Server's listen method.
+    this._server = this.expressServer.listen(
+      (this.config && this.config.port) || 3000
+    );
+    return Promise.resolve();
+  }
+  async stop(): Promise<void> {
+    this._server.close();
+    return Promise.resolve();
+  }
+}
+
+export type RPCServerConfig = {
+  port?: number;
+  [key: string]: any;
+};
+```
+
+### Putting it all together
+Now here is the Application subclass and `index.ts` entry point to
+make it go!
+
+{% include code-caption.html content="/src/application.ts" %}
+```ts
+import {Application, ApplicationConfig} from '@loopback/core';
+import {RPCServer} from './servers/rpc-server';
+import {GreetController} from './controllers';
+
+export class MyApplication extends Application {
+  options: ApplicationConfig;
+  constructor(options?: ApplicationConfig) {
+    // Allow options to replace the defined components array, if desired.
+    super(options);
+    this.controller(GreetController);
+    this.server(RPCServer);
+    this.options = options || {};
+    this.options.port = this.options.port || 3000;
+    this.bind('rpcServer.config').to(this.options);
+  }
+
+  async start() {
+    await super.start();
+    console.log(`Server is running on port ${this.options.port}`);
+  }
+}
+```
+
+{% include code-caption.html content="index.ts" %}
+```ts
+import {MyApplication} from './src/application';
+
+const app = new MyApplication();
+
+app.start().catch(err => {
+  console.error(`Unable to start application: ${err}`);
+});
+async function shutdown() {
+  await app.stop();
+  process.exit(0);
+}
+// Event handlers for when we are killed; these aren't *required*.
```

Comment:
Well, if we want to keep our examples in this page short and readable, then I think this section should be removed, since it's not really needed? If you want to show how to use stop, then perhaps use an end-to-end test as the example and call start/stop from before/after hooks.